core/src/main/java/org/mule/runtime/core/el/DataWeaveExpressionLanguage.java

```diff
@@ -123,14 +139,19 @@ private void addEventBindings(Event event, BindingContext.Builder contextBuilder
         contextBuilder.addBinding(name, value);
       });
       contextBuilder.addBinding(VARIABLES,
-                                new TypedValue(unmodifiableMap(flowVars), fromType(flowVars.getClass())));
+                                new TypedValue<>(unmodifiableMap(flowVars), fromType(flowVars.getClass())));
       Message message = event.getMessage();
       Attributes attributes = message.getAttributes();
-      contextBuilder.addBinding(ATTRIBUTES, new TypedValue(attributes, fromType(attributes.getClass())));
+      contextBuilder.addBinding(ATTRIBUTES, new TypedValue<>(attributes, fromType(attributes.getClass())));
       contextBuilder.addBinding(PAYLOAD, message.getPayload());
-      contextBuilder.addBinding(DATA_TYPE, new TypedValue(message.getPayload().getDataType(), fromType(DataType.class)));
+      contextBuilder.addBinding(DATA_TYPE, new TypedValue<>(message.getPayload().getDataType(), fromType(DataType.class)));
+      contextBuilder.addBinding(MULE,
```

Comment:
This means the mule, server and app variables will be added per Event but they are not related. They should be global bindings instead.